Core/AMPCanonicalExtractor.swift

```diff
@@ -0,0 +1,159 @@
+//
+//  AMPCanonicalExtractor.swift
+//  DuckDuckGo
+//
+//  Copyright © 2021 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import WebKit
+
+public class AMPCanonicalExtractor: NSObject {
+    
+    public static let shared = AMPCanonicalExtractor()
```

Comment:
We could move it to TabVC and have it as a property there, so lifetime and state is encapsulated and "connected" to state of the Tab itself.